src/App.tsx

```diff
@@ -1,14 +1,95 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { FilterMode } from './types/FilterMode';
+import { TodoModalInfo } from './types/TodoModalInfo';
+import { User } from './types/User';
+import { getTodos, getUser } from './api';
+
+let fetchedTodos: Todo[] = [];
+const initialModalInfo: TodoModalInfo = {
+  todoId: 0,
+  todoTitle: '',
+  todoCompleted: false,
+  userEmail: '',
+  userName: '',
+  showModalLoader: true,
+};
 
 export const App: React.FC = () => {
+  const [displayedTodos, setDisplayedTodos] = useState<Todo[]>([]);
+  const [todosLoaded, setTodosLoaded] = useState(false);
+  const [showModal, setShowModal] = useState(false);
+  const [modalInfo, setModalInfo] = useState<TodoModalInfo>(initialModalInfo);
+  const [clickedTodoId, setClickedTodoId] = useState(0);
```

Comment:
I recommend you try simplifying this logic to use only one state (like `selectedTodo`)